controllers/argocd/notifications/role.go

```diff
@@ -0,0 +1,151 @@
+package notifications
+
+import (
+	"reflect"
+
+	"github.com/argoproj-labs/argocd-operator/pkg/argoutil"
+	"github.com/argoproj-labs/argocd-operator/pkg/cluster"
+	"github.com/argoproj-labs/argocd-operator/pkg/mutation"
+	"github.com/argoproj-labs/argocd-operator/pkg/permissions"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (nr *NotificationsReconciler) reconcileRole() error {
+
+	nr.Logger.Info("reconciling roles")
+
+	roleRequest := permissions.RoleRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        argoutil.GenerateUniqueResourceName(nr.Instance.Name, nr.Instance.Namespace, ArgoCDNotificationsControllerComponent),
+			Namespace:   nr.Instance.Namespace,
+			Labels:      nr.Instance.Labels,
```

Comment:
we should call util.DefaultLabels here to get all the labels we need

Reply:
This is set to a global value already in the `reconcile()` function in `notifications.go`.

Reply:
Once your PR gets merged, I will update it to `util.DefaultLabels()` instead.